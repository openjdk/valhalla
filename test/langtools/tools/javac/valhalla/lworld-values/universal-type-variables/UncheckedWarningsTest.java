/*
 * @test /nodynamiccopyright/
 * @summary unchecked warning test for universal type variables
 * @compile/ref=UncheckedWarningsTest.out -Xlint:unchecked -XDrawDiagnostics UncheckedWarningsTest.java
 */

class UncheckedWarningsTest {
    static primitive class Atom { }

    static class Box<__universal X> { }
    static class Pair<__universal X, __universal Y> { }
    static class Triple<__universal X, __universal Y, __universal Z> { }

    public static void main(String[] args) {
        Box<Box<Box<Atom>>> val = null;
        Box<Box<Box<Atom.ref>>> ref = null;
        val = ref;
    }
}
