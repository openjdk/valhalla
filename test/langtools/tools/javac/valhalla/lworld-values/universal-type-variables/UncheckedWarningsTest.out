UncheckedWarningsTest.java:17:15: compiler.warn.prob.found.req: (compiler.misc.unchecked.assign), UncheckedWarningsTest.Box<UncheckedWarningsTest.Box<UncheckedWarningsTest.Box<UncheckedWarningsTest.Atom.ref>>>, UncheckedWarningsTest.Box<UncheckedWarningsTest.Box<UncheckedWarningsTest.Box<UncheckedWarningsTest.Atom>>>
1 warning
